### PR TITLE
[new downloader] Getting a list of parents

### PR DESCRIPTION
**Pull request by bykoianko**
Получение списка родителей и тесты.

https://jira.mail.ru/browse/MAPSME-81

### CODE REVIEW

File: storage/storage.cpp [hunk 1]
```diff
@@ -1153,11 +1158,19 @@ void Storage::GetNodeAttrs(TCountryId const & countryId, NodeAttrs & nodeAttrs)
...
   nodeAttrs.m_error = statusAndErr.error;
   nodeAttrs.m_nodeLocalName = m_countryNameGetter(countryId);
-  nodeAttrs.m_parentCountryId = nodeValue.GetParent();
-  if (nodeAttrs.m_parentCountryId.empty())
-    nodeAttrs.m_parentLocalName.clear();
-  else
-    nodeAttrs.m_parentLocalName = m_countryNameGetter(nodeAttrs.m_parentCountryId);
+
+  nodeAttrs.m_parentInfo.clear();
+  for (auto const & n : nodes)
```
Comment (kshalnev) [inline storage/storage.cpp]: nodeAttrs.m_parentInfo.reserve(nodes.size());

File: storage/country.hpp [hunk 2]
```diff
@@ -25,7 +25,16 @@ namespace storage
 {
 using TMapping = map<TCountryId, TCountriesSet>;
 
-/// Serves as a proxy between GUI and downloaded files
+/// This class keep all the information about a country in countre tree (TCountriesContainer).
```
Comment (kshalnev) [inline storage/country.hpp]: keeps

File: storage/country.hpp [hunk 3]
```diff
@@ -25,7 +25,16 @@ namespace storage
...
+/// It is guaranteed that every node represent a unique region has a unique |m_name| in country tree.
+/// If several nodes have the same |m_name| they represents the same region.
```
Comment (kshalnev) [inline storage/country.hpp]: represent

File: storage/country.hpp [hunk 4]
```diff
@@ -25,7 +25,16 @@ namespace storage
...
+/// It could happend in case of disputed territories.
```
Comment (kshalnev) [inline storage/country.hpp]: It happens in case

File: storage/storage.cpp [hunk 5]
```diff
@@ -244,7 +244,7 @@ size_t Storage::GetDownloadedFilesCount() const
 TCountriesContainer const & NodeFromCountryId(TCountriesContainer const & root,
                                               TCountryId const & countryId)
 {
-  SimpleTree<Country> const * node = root.FindLeaf(Country(countryId));
+  SimpleTree<Country> const * node = root.FindFirstLeaf(Country(countryId));
```
Comment (kshalnev) [inline storage/storage.cpp]: rename func to LeafNodeFromCountryId?

**Comment by kshalnev** [comment]
LGTM after minor fixes

File: storage/storage.cpp (hunk 1 above)
Comment (bykoianko) [inline storage/storage.cpp]: DONE

File: storage/country.hpp (hunk 2 above)
Comment (bykoianko) [inline storage/country.hpp]: Done

File: storage/country.hpp (hunk 3 above)
Comment (bykoianko) [inline storage/country.hpp]: Fixed

File: storage/country.hpp (hunk 4 above)
Comment (bykoianko) [inline storage/country.hpp]: Fixed

File: storage/storage.cpp (hunk 5 above)
Comment (bykoianko) [inline storage/storage.cpp]: I cann't do that. SimpleTree is a template class. Here it's used with param Country.

Comment (bykoianko) [inline storage/storage.cpp]: DONE

**Pull request merged**